src/include/random.h

```diff
@@ -133,13 +133,14 @@ void randomize_rng()
   detail::randomize_rng<EngineT>();
 }
 
-template <int min = 0,
-          int max = std::numeric_limits<int>::max(), 
-          typename DistributionT = std::uniform_int_distribution<int>,
+template <typename IntegerT,
+          IntegerT min = std::numeric_limits<IntegerT>::min(),
+          IntegerT max = std::numeric_limits<IntegerT>::max(),
+          typename DistributionT = std::uniform_int_distribution<IntegerT>,
           typename EngineT = std::default_random_engine>
-int generate_random_number()
+IntegerT generate_random_number()
 {
-  return detail::generate_random_number<int, DistributionT, EngineT>
+  return detail::generate_random_number<IntegerT, DistributionT, EngineT>
```

Comment:
Can the compiler deduce that..? Otherwise all calls will have to look like:
generate_random_number<int>()?
...which I don't think we want. Ok if the default is "int" and we use the explicit parameter for special cases, though (that was the original intent).

Reply:
Could we have the generic version and make a few aliases?

generate_random_int, generate_random_long…

Reply:
We could, but I think that if int is indeed the type we want the most, it's best to just leave that as default and just use explicit specialization for when we need them. There's also no reason that in any particular module one couldn't write:
 auto gen_double = [](double min, double max) {
                        return ceph::util::generate_random_number<double>(min, max);
                   };
...or something. Unless you feel like such aliases would be really frequently useful?

Reply:
Well, I tend to rarely use straight-up int when I'm writing code. If I were making a library for myself I'd mostly just take the generic version because I'll always want a uint32_t or a uint16_t or an int64_t or…

Reply:
One of the reasons for that design choice (and this also is why min defaults to 0) is that the main purpose of the library is to be a drop-in replacement for rand(), which implies int and (0-max]. But, if we're finding ourselves needing a specific size all the time, having alternatives with certain sizes or different ranges starts to make sense; I see no reason we couldn't add them, but I don't think they'd be right for replacing the main interface since they'd break the contract.